File: 45-jump-game-ii/jump-game-ii.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int jumps = 0;
        int currentEnd=0;
        int maxReach = 0;
        for(int i=0;i<nums.size()-1;i++){
            if (maxReach<nums[i]+i){maxReach = nums[i]+i;}
            if(i==currentEnd){
                jumps++;
                currentEnd=maxReach;
            }
        }
        return jumps;
        
    }
};

        

File: 45-jump-game-ii/jump-game-ii.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        jumps = 0
        currentEnd = 0
        maxReach = 0
        
        for i in range(len(nums) - 1):
            maxReach = max(maxReach, i + nums[i])
            
            if i == currentEnd:
                jumps += 1
                currentEnd = maxReach

        return jumps

        